dimscord/helpers/channel.nim

```diff
@@ -59,39 +58,35 @@ template create*(ch: GuildChannel, what: typedesc[Invite];
         target_type, reason
     )
 
-template delete*(inv: Invite, reason = ""): Future[void] =
+template delete*(inv: Invite | string, reason = ""): Future[void] =
     ## Delete a guild invite.
-    getClient.api.deleteInvite(inv.code, reason)
+    let code = when inv is Invite: inv.code else: inv
+    getClient.api.deleteInvite(code, reason)
 
 template invites*(ch: GuildChannel): Future[seq[Invite]] =
     ## Gets a list of a channel's invites.
     getClient.api.getChannelInvites(ch.id)
 
-template channel*(g: Guild, channel_id: string): Future[GuildChannel] =
-    ## Gets guild channel by ID.
-    getClient.api.getChannel(channel_id)
-
-template dms*(s: Shard, channel_id: string): Future[DMChannel] =
-    ## Gets dm channel by ID
-    let chan = getClient.api.getChannel(channel_id)
-    if chan[1].isSome:
-        get chan[1]
-
-    case result.kind
-    of ctDirect, ctGroupDM:
-        discard
-    else: 
-        raise newException(CatchableError, "Channel type is not a dm")
+template getChannel*(g: Guild, channel_id: string): Future[SomeChannel] =
+    ## Gets channel by ID.
+    let channel = getClient.api.getChannel(channel_id)
+    if channel[0].isSome:
+        result = channel[0].get
+    elif channel[1].isSome:
+        result = channel[1].get
```

Comment:
This won't compile since the return type now depends on runtime information.
Also templates don't have a `result` variable so this would be the `result` variable of where the template is called from